clarkson-objects/Clarkson_Archive_Object.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Clarkson Archive.
+ *
+ * @package CLARKSON\Objects
+ */
+
+/**
+ * Clarkson Archive class.
+ */
+class Clarkson_Archive_Object {
+
+	/**
+	 * Define $_post_type.
+	 *
+	 * @var WP_Post_Type|null
+	 */
+	protected $_type;
+
+	/**
+	 * Clarkson_Archive_Object constructor.
+	 *
+	 * @param string|null $type Post type.
+	 *
+	 * @throws Exception Error message.
+	 */
+	public function __construct( $type = null ) {
+		if ( ! $type ) {
+			throw new Exception( $type . ' empty' );
+		}
+		$this->_type = get_post_type_object( $type );
+		if ( ! $this->_type ) {
+			throw new Exception( 'Post type not found' );
+		}
+		if ( ! $this->_type->has_archive ) {
+			throw new Exception( 'Post type has no archive' );
+		}
+	}
+
+	/**
+	 * Check if an existing field is requested.
+	 *
+	 * @param string $name Field to search by.
+	 *
+	 * @throws Exception Error message.
+	 */
+	public function __get( $name ) {
+		throw new Exception( 'Trying to access WordPress Archive object properties while WordPress doesn\'t have an archive object.' );
+	}
+
+	/**
+	 * Get the post type data.
+	 *
+	 * @return string|null
+	 */
+	public function get_title() {
+		$title = apply_filters( 'post_type_archive_title', $this->_type->labels->name, $this->_type->name );
```

Comment:
Maybe use `get_the_archive_title` instead of some custom logic? https://developer.wordpress.org/reference/functions/get_the_archive_title/

Reply:
I'm wondering why I didn't find this myself... Thanks :)

Reply:
Well I just tried, and there is a specific difference between the 2 methods.

When using the built in `get_the_archive_title` function, for custom post types this will return a string with `Archives: POST_TYPE_NAME`. I think that's not the expected behavior.
Result: see https://prnt.sc/ooy6mo

My 'custom' logic is actually based on this: https://developer.wordpress.org/reference/functions/post_type_archive_title/
But I stripped out the functionality we don't need or has already been done to prevent extra database calls (like `get_post_type_object()`).
Result: see https://prnt.sc/ooy84v

So I guess there are 2 options here, which result in a different title.

Reply:
Totally understand 👍

Reply:
Totally understand 👍  Normally I'm not a big fan of copying code from core, but in this case you now also can calls this Post_Type object on a single CPT page. When using the WP core function it returns early with nothing because of the `is_post_type_archive` check.